src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp

```diff
@@ -0,0 +1,132 @@
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace GeneralizedHarmonic::BoundaryConditions {
+/*!
+ * \brief Sets Dirichlet boundary conditions using the analytic solution or
+ * analytic data.
+ */
+template <size_t Dim>
+class DirichletMinkowski final : public BoundaryCondition<Dim> {
+ public:
+  using options = tmpl::list<>;
+  static constexpr Options::String help{
+      "DirichletMinkowski boundary conditions setting the value of the "
+      "spacetime metric and its derivatives Phi and Pi to Minkowski (i.e., "
+      "flat spacetime)"};
+
+  DirichletMinkowski() = default;
+  DirichletMinkowski(DirichletMinkowski&&) noexcept = default;
+  DirichletMinkowski& operator=(DirichletMinkowski&&) noexcept = default;
+  DirichletMinkowski(const DirichletMinkowski&) = default;
+  DirichletMinkowski& operator=(const DirichletMinkowski&) = default;
+  ~DirichletMinkowski() override = default;
+
+  explicit DirichletMinkowski(CkMigrateMessage* msg) noexcept;
+
+  WRAPPED_PUPable_decl_base_template(
+      domain::BoundaryConditions::BoundaryCondition, DirichletMinkowski);
+
+  auto get_clone() const noexcept -> std::unique_ptr<
+      domain::BoundaryConditions::BoundaryCondition> override;
+
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::Ghost;
+
+  void pup(PUP::er& p) override;
+
+  using dg_interior_evolved_variables_tags = tmpl::list<>;
+  using dg_interior_temporary_tags = tmpl::list<
+      domain::Tags::Coordinates<Dim, Frame::Inertial>,
+      ::GeneralizedHarmonic::ConstraintDamping::Tags::ConstraintGamma1,
+      ::GeneralizedHarmonic::ConstraintDamping::Tags::ConstraintGamma2>;
+  using dg_gridless_tags = tmpl::list<::Tags::Time>;
+
+  std::optional<std::string> dg_ghost(
+      const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+          spacetime_metric,
+      const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*> pi,
+      const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*> phi,
+      const gsl::not_null<Scalar<DataVector>*> gamma1,
+      const gsl::not_null<Scalar<DataVector>*> gamma2,
+      const gsl::not_null<Scalar<DataVector>*> lapse,
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> shift,
+      const gsl::not_null<tnsr::II<DataVector, Dim, Frame::Inertial>*>
+          inv_spatial_metric,
+      const std::optional<
+          tnsr::I<DataVector, Dim, Frame::Inertial>>& /*face_mesh_velocity*/,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& /*normal_covector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+      const Scalar<DataVector>& interior_gamma1,
+      const Scalar<DataVector>& interior_gamma2,
+      const double time) const noexcept {
+    *gamma1 = interior_gamma1;
+    *gamma2 = interior_gamma2;
+    auto boundary_values = [&coords, &time]() noexcept {
```

Comment:
just set everything here to their analytical values without calling into the analytic solution, or using the other free functions.  Then you don't need to specify the coordinates or time in the tags above or take them as arguments